docs: ACLs default deny, service intentions k8s prerequisite (#13743)
hashicorp/consul: website/content/docs/connect/transparent-proxy.mdx

> * To use transparent proxy on Kubernetes, Consul-helm >= `0.32.0` and Consul-k8s >= `0.26.0` are required in addition to Consul >= `1.10.0`.

Reviewer: ```suggestion
* Consul Helm chart 0.32.0 or later 
* Consul on Kubernetes 0.26.0 r later
* Consul 1.10.0 or later
```
There are other issues with how this is organized, but let's address some low-hanging fruit and use some conventional language while we're at it.

> * If the default policy for ACLs is "deny", then Service Intentions should be set up to allow intended services to connect to each other.

Reviewer: ```suggestion
* If your default ACLs policy is `deny`, you must configure service intentions to explicitly allow services to connect to each other.
```
Use lower case for service intentions because it is a regular noun.
Use active voice.

> Otherwise, Connect services will not be able to talk as the default Service Intentions policy will be inherited from the default ACL policy.

Reviewer: ```suggestion
Services in the mesh inherit the default ACL policy and will not be able to communicate if the default policy is to deny.
```
Is that any cleaner given the suggestion in the previous line?

> * If using Transparent Proxy, all worker nodes within a Kubernetes cluster must have the `ip_tables` kernel module running, e.g. `modprobe ip_tables`. 

Reviewer: ```suggestion
* All worker nodes within a Kubernetes cluster must have the `ip_tables` kernel module running, e.g. `modprobe ip_tables`. 
```
I think the context suggests that the reader is using transparent proxy.
